### PR TITLE
Save message body as file

### PR DESCRIPTION
**Pull request by evilbaschdi**
As discussed in issue [358](https://github.com/paolosalvatori/ServiceBusExplorer/issues/358) I've extended the code to support saving a messages body content to a file. Also bulk operations are supported to save more than one message body at once.

### CODE REVIEW

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-PullRequest0378.128.build.588 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/27458047) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/61610dfd28 by @evilbaschdi)

File: src/ServiceBusExplorer.sln.DotSettings [hunk 1]
```diff
@@ -590,6 +590,9 @@ II.2.12 &lt;HandlesEvent /&gt;&#xD;
 	<s:String x:Key="/Default/CodeStyle/Naming/XamlNaming/UserRules/=XAML_005FFIELD/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/XamlNaming/UserRules/=XAML_005FRESOURCE/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
 	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpFileLayoutPatternsUpgrade/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpKeepExistingMigration/@EntryIndexedValue">True</s:Boolean>
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer.sln.DotSettings]: Probably shouldn't be part of this PR.

File: src/ServiceBusExplorer/Controls/HandleQueueControl.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@
 // This sample is supplemental to the technical guidance published on my personal
 // blog at http://blogs.msdn.com/b/paolos/. 
 // 
-// Author: Paolo Salvatori
+// Author: Paolo Salvatori, Sebastian Walter
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer/Controls/HandleQueueControl.cs]: @paolosalvatori FYI

File: src/ServiceBusExplorer/Controls/HandleQueueControl.cs [hunk 3]
```diff
@@ -307,34 +329,18 @@ public partial class HandleQueueControl : UserControl
         #region Private Static Fields
 
         private static readonly List<string> ClaimTypes = new List<string>
-        {
-            "NameIdentifier",
-            "Upn",
-            "Role",
-            "SharedAccessKey"
-        };
+                                                          {
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer/Controls/HandleQueueControl.cs]: This should not be part of this PR.

File: src/ServiceBusExplorer/Controls/HandleQueueControl.cs [hunk 4]
```diff
@@ -21,30 +21,49 @@
...
-using Microsoft.Azure.ServiceBusExplorer.Forms;
-using Microsoft.Azure.ServiceBusExplorer.Helpers;
-
-#endregion
 
 // ReSharper disable once CheckNamespace
 
 namespace Microsoft.Azure.ServiceBusExplorer.Controls
 {
+    #region Using Directives
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer/Controls/HandleQueueControl.cs]: This should not be part of this PR.

File: src/ServiceBusExplorer/Controls/HandleQueueControl.cs [hunk 5]
```diff
@@ -529,18 +540,18 @@ public void GetMessageSessions()
 
                 var messageSessions = sessionEnumerable as MessageSession[] ?? sessionEnumerable.ToArray();
                 sessionBindingList = new SortableBindingList<MessageSession>(messageSessions)
-                {
-                    AllowEdit = false,
-                    AllowNew = false,
-                    AllowRemove = false
-                };
+                                     {
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer/Controls/HandleQueueControl.cs]: This should not be part of this PR.

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-PullRequest0378.130.build.589 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/27486367) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/4d69d2f440 by @evilbaschdi)

**Comment by evilbaschdi** [comment]
I reversed some hunks and did a "git push --force" ... hopefully this works

**Comment by SeanFeldman** [comment]
Not everything got reverted properly. Do you want to send a fresh PR to ensure only the vital changes are in it?

**Comment by evilbaschdi** [comment]
Sure I can. But I had a quick look and see the issues. I'm sorry but it seems SourceTree has rolled back some changes I did since it was crashing all the time yesterday morning. So it's also possible to revert the rest of those "wrong" changes and push in this branch again.

**Comment by SeanFeldman** [comment]
No worries, happens.
I suggested a new PR (branch can remain the same) to have a clean work slate.
Marking this as WIP. Once you're done, remove WIP and ping us. Thanks.

**Comment by evilbaschdi** [comment]
thanks to https://www.devart.com/codecompare/ we have another try :-)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-PullRequest0378.132.build.594 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/27590233) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/b6feb11838 by @evilbaschdi)

**Comment by SeanFeldman** [comment]
It's better now 🙂.
Let's coordinate with @ErikMogensen to ensure the changes are not conflicting with his work in PR #381.

**Comment by evilbaschdi** [comment]
Last time I had a look into his branch @ErikMogensen did not make changes to the files contained in my PR so I don't think there should be bigger merge issues.

**Comment by SeanFeldman** [comment]
Sounds good. There will be a 2nd reviewer anyways, so if I missed anything big, it will be (hopefully) caught.
